pkg/dynamic/custom_reconcile.go

```diff
@@ -238,6 +238,19 @@ func MergeMaps(m1, m2 map[string]string) map[string]string {
 	return merged
 }
 
+// mergeMaps merges the values in the passed maps into a new map.
+// Values within m2 potentially clobber m1 values.
+func mergeMaps(m1, m2 map[string]interface{}) map[string]interface{} {
```

Comment:
small nit: we already have a function called MergeMaps, so maybe call this mergeStringInterfaceMaps (or call the other one mergeStringMaps)?